galsim/wfirst/wfirst_backgrounds.py

```diff
@@ -0,0 +1,175 @@
+    # The table only includes longitude in the range [0, 180] because there is symmetry in that a
+    # negative longitude in the range[-180, 0] should have the same sky level as at the positive
+    # value of longitude (given that the Sun is at 0).
+    if position is None:
+        if e_lat is None:
+            ecliptic_lat = 30.
+            ecliptic_lon = 90.
+        else:
+            ecliptic_lat = e_lat / galsim.degrees
+            ecliptic_lon = e_lon / galsim.degrees
+    else:
+        ecliptic_lon, ecliptic_lat = position.ecliptic()
+        ecliptic_lon = ecliptic_lon / galsim.degrees
+        ecliptic_lat = ecliptic_lat / galsim.degrees
+    # Here's where we take advantage of longitude symmetries:
+    if ecliptic_lon > 180.:
+        ecliptic_lon -= 360.
+    ecliptic_lon = abs(ecliptic_lon)
+    # And latitude symmetries:
+    if ecliptic_lat < 0.:
+        ecliptic_lat = abs(ecliptic_lat)
+    sin_ecliptic_lat = np.sin(np.pi*ecliptic_lat/180.)
+
+    # Take the lookup table, and turn negative numbers (indicating failure because of proximity to
+    # sun) to large positive values so that we can identify them as bad after interpolation.
+    max_sky = np.max(bandpass._sky_level)
+    sky_level = bandpass._sky_level.copy()
+    sky_level[sky_level<0] = 1.e6
+
+    # Interpolate!  We could do a full 2d interpolation, but this seems a bit like overkill.  We're
+    # going to first interpolate in latitude, then in longitude.
+    # Equal spacing in sin(latitude).
+    all_sin_lat_vals = np.sin(np.pi*bandpass._ecliptic_lat/180.)
+    sin_lat_vals = np.sort(np.array(list(set(all_sin_lat_vals))))
+    n_sl = len(sin_lat_vals)
+    d_sl = sin_lat_vals[1]-sin_lat_vals[0]
+    # Handle edge cases:
+    if sin_ecliptic_lat <= sin_lat_vals[0]:
+        sky_table_1d = sky_level[all_sin_lat_vals == sin_lat_vals[0]]
+    elif sin_ecliptic_lat >= sin_lat_vals[n_sl-1]:
+        sky_table_1d = sky_level[all_sin_lat_vals == sin_lat_vals[n_sl-1]]
+    else:
+        interp_val = (sin_ecliptic_lat - sin_lat_vals[0])/d_sl
+        lower_ind = int(np.floor(interp_val))
+        frac = interp_val - lower_ind
+        sky_table_1d = (1.-frac)*sky_level[all_sin_lat_vals == sin_lat_vals[lower_ind]] \
+            + frac*sky_level[all_sin_lat_vals == sin_lat_vals[lower_ind+1]]
+    # Now we have a table to interpolate on the set of galactic longitude values.
+    long_vals = np.sort(np.array(list(set(bandpass._ecliptic_lon))))
+    n_l = len(long_vals)
+    d_l = long_vals[1]-long_vals[0]
+    # Handle edge cases:
+    if ecliptic_lon <= long_vals[0]:
+        sky_val = sky_table_1d[0]
+    elif ecliptic_lon >= long_vals[n_l-1]:
+        sky_val = sky_table_1d[n_l-1]
+    else:
+        interp_val = (ecliptic_lon - long_vals[0])/d_l
+        lower_ind = int(np.floor(interp_val))
+        frac = interp_val - lower_ind
+        sky_val = (1.-frac)*sky_table_1d[lower_ind] + frac*sky_table_1d[lower_ind]
```

Comment:
Now, perhaps a more important point about this.  I'm pretty sure the "ecliptic" longitude is not the normal ecliptic coordinate system.  I'm pretty sure this table considers the sun to be at lat, long = 0,0.  That's why the numbers shoot up there and get undefined when you get too close to 0,0.  So I think this routine needs some kind of date of observation to calculate what ecliptic longitude the sun is for this observation.  Right now I believe it is implicitly assuming that the observations are taking place on March 21.